cpp/models/lct_secir/infection_state.h

```diff
@@ -86,29 +86,35 @@ class InfectionState
      * @brief Constructor for the InfectionState class.
      *
      * InfectionState class defines the possible InfectionState%s with the number of Subcompartments for the LCT model.
-     * @param[in] SubcompartmentNumbers Vector which defines the number of Subcompartments for each infection state of InfectionStateBase.       
+     * @param[in] subcompartment_numbers Vector which defines the number of Subcompartments for each infection state of InfectionStateBase.       
      */
-    InfectionState(std::vector<int> SubcompartmentNumbers)
-        : m_SubcompartmentNumbers(std::move(SubcompartmentNumbers))
-        , m_SubcompartmentNumbersindexfirst(std::vector<int>((int)InfectionStateBase::Count, 1))
+    InfectionState(std::vector<int> subcompartment_numbers)
+        : m_subcompartment_numbers(std::move(subcompartment_numbers))
+        , m_subcompartment_indexfirst((int)InfectionStateBase::Count, 1)
     {
-        bool constraint_check = check_constraints();
-        if (!constraint_check) {
-            set_compartment_index();
-        }
+        check_constraints();
+        set_compartment_index();
     }
 
     /**
      * @brief Setter for the number of Subcompartments.
      *
-     * @param[in] SubcompartmentNumbers Vector which defines the number of Subcompartments for each infection state of InfectionStateBase.       
+     * The number of Subcompartments is only updated if the vector is valid.
+     * @param[in] subcompartment_numbers Vector which defines the number of Subcompartments for each infection state of InfectionStateBase. 
+     * @return Returns true if the vector is not valid, otherwise false.      
      */
-    void set_SubcompartmentNumbers(std::vector<int> SubcompartmentNumbers)
+    bool set_subcompartment_numbers(std::vector<int> subcompartment_numbers)
     {
-        m_SubcompartmentNumbers = SubcompartmentNumbers;
-        bool constraint_check   = check_constraints();
-        if (!constraint_check) {
+        std::vector<int> copy_m_subcompartment_numbers(m_subcompartment_numbers);
+        m_subcompartment_numbers = std::move(subcompartment_numbers);
+        if (check_constraints()) {
```

Comment:
Intuitively, it would make more sense to me if the check_constraint() function is true when everything is ok. But if you want to leave it like this, you can simply resolve this comment.
´

Reply:
Basically I agree with you here, but the function was implemented the same way in the IDE and OD models. That's why I wanted to keep the consistency here. Maybe this could be changed for all models, but given the description in the function documentation, I think it's actually okay.